File: udp_server.rb
require 'socket'


class UDPServer
  def initialize host = 'raspberrypi.local', port = 2200
    @socket = UDPSocket.new
    @socket.bind host, port
  end

  def read
    reply, from = @socket.recvfrom 20
    reply
  end
end


if __FILE__ == $0
  server = UDPServer.new
  while true
    puts server.read
  end
end

File: udp_server.py
import socket


class UDPServer:
    def __init__(self, host = '', port = 2200, timeout = 0.0):
        self.socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
        self.socket.bind((host, port))
        self.socket.settimeout(timeout)

    def read(self):
        try:
            reply, addr = self.socket.recvfrom(64)
            return reply
        except socket.error:
            pass


if __name__ == "__main__":
    server = UDPServer('', 2200)
    while True:
        print(server.read())
